hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -462,6 +467,15 @@ Connection connect(Object target, Function<Object, Connection> getOrConnectFunct
         } catch (ClientNotAllowedInClusterException e) {
             logger.warning("Exception during initial connection to " + target + ": " + e);
             throw e;
+        } catch (TargetDisconnectedException e) {
+            connectionProcessListener.remoteClosedConnection(addressProvider.apply(target));
+            return null;
+        } catch (HazelcastException e) {
+            logger.warning("Exception during initial connection to " + target + ": " + e);
+            if (e.getCause() instanceof IOException) {
+                connectionProcessListener.connectionAttemptFailed(addressProvider.apply(target));
+            }
```

Comment:
When can we hit this line? Where do we wrap IOException with a HazelcastException?

Reply:
`addressProvider.apply(target)` can possibly throw. should we handle it?

Reply:
Updated with a try-catch-log.